x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/allocation/TrainedModelAllocationNodeService.java

```diff
@@ -168,6 +178,20 @@ public void stop() {
 
     void loadQueuedModels() {
         TrainedModelDeploymentTask loadingTask;
+        if (latestState != null) {
```

Comment:
Do a cheap check that there is something to load before checking the indices. This method is called on a schedule

```suggestion
        if (loadingModels.isEmpty()) {
            return;
        }
```